### PR TITLE
Replace GCN use of **kwargs with real arguments

### PR DESCRIPTION
**Pull request by huonw**
The headers for the `GCN` and `GraphConvolution` classes now shows all the information about the class:

`class stellargraph.layer.gcn.GCN(layer_sizes, generator, bias=True, dropout=0.0, activations=None, kernel_initializer=None, kernel_regularizer=None, kernel_constraint=None, bias_initializer=None, bias_regularizer=None, bias_constraint=None)`

`class stellargraph.layer.gcn.GraphConvolution(units, activation=None, use_bias=True, final_layer=False, input_dim=None, kernel_initializer='glorot_uniform', kernel_regularizer=None, kernel_constraint=None, bias_initializer='zeros', bias_regularizer=None, bias_constraint=None, **kwargs)`

Especially for GCN, this will reduce the chance of misspelled arguments being silently ignored.

See: #801

### CODE REVIEW

File: stellargraph/layer/gcn.py [hunk 1]
```diff
@@ -57,46 +57,45 @@ class GraphConvolution(Layer):
...
+        kernel_regularizer=None,
+        kernel_constraint=None,
+        bias_initializer="zeros",
+        bias_regularizer=None,
+        bias_constraint=None,
+        **kwargs
     ):
-        if "input_shape" not in kwargs and "input_dim" in kwargs:
-            kwargs["input_shape"] = (kwargs.get("input_dim"),)
+        if "input_shape" not in kwargs and input_dim is not None:
```
Comment (kjun9) [inline stellargraph/layer/gcn.py]: Why does `input_shape` get special treatment and is still part of kwargs? Is it purely because of this parameter that we can't remove `**kwargs` from the argument list altogether?

Comment (huonw) [reply stellargraph/layer/gcn.py]: No, `input_shape` is a kwarg to Keras's `Layer` (https://github.com/tensorflow/tensorflow/blob/a641fa1c5c10f0a278c7671fd6f7df550a74935d/tensorflow/python/keras/engine/base_layer.py#L278-L291
) (along with other kwargs), and it's typical for Keras layer subclasses to have a `**kwargs` to pass through this and anything else, e.g.: https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense 's `__init__` has a `**kwargs` too.

Most of the stellargraph layers seem to have an `input_dim` argument to make defining single-dimensional input slightly easier, I guess.
